ui/perfherder/compare/CompareTableView.jsx

```diff
@@ -0,0 +1,370 @@
+      framework: this.getFrameworkData(),
+      filteredResults: new Map(),
+      filterOn: false,
+      title: '',
+    };
+  }
+
+  componentDidMount() {
+    this.getPerformanceData();
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props !== prevProps) {
+      this.getPerformanceData();
+    }
+  }
+
+  getFrameworkData = () => {
+    const { framework, frameworks } = this.props.validated;
+
+    if (framework) {
+      const frameworkObject = frameworks.find(
+        item => item.id === parseInt(framework, 10),
+      );
+      // framework is validated in the withValidation component so
+      // we know this object will always exist
+      return frameworkObject;
+    }
+    return { id: 1, name: 'talos' };
+  };
+
+  setTimeRange = () => {
+    const { selectedTimeRange, originalRevision } = this.props.validated;
+
+    if (originalRevision) {
+      return null;
+    }
+
+    let timeRange;
+    if (selectedTimeRange) {
+      timeRange = phTimeRanges.find(
+        timeRange => timeRange.value === parseInt(selectedTimeRange, 10),
+      );
+    }
+
+    return timeRange || compareDefaultTimeRange;
+  };
+
+  getPerformanceData = async () => {
+    const { getQueryParams, hasSubtests, getDisplayResults } = this.props;
+    const {
+      originalProject,
+      originalRevision,
+      newProject,
+      newRevision,
+    } = this.props.validated;
+    const { framework, timeRange } = this.state;
+
+    this.setState({ loading: true });
+
+    const [originalParams, newParams] = getQueryParams(timeRange, framework);
+
+    const [originalResults, newResults] = await Promise.all([
+      getData(createApiUrl(perfSummaryEndpoint, originalParams)),
+      getData(createApiUrl(perfSummaryEndpoint, newParams)),
+    ]);
+
+    if (originalResults.failureStatus) {
+      return this.setState({
+        failureMessage: originalResults.data,
+        loading: false,
+      });
+    }
+
+    if (newResults.failureStatus) {
+      return this.setState({
+        failureMessage: newResults.data,
+        loading: false,
+      });
+    }
+
+    const data = [...originalResults.data, ...newResults.data];
+    let rowNames;
+    let tableNames;
+    let title;
+
+    if (!data.length) {
+      return this.setState({ loading: false });
+    }
+
+    if (hasSubtests) {
+      let subtestName = data[0].name.split(' ');
+      subtestName.splice(1, 1);
+      subtestName = subtestName.join(' ');
+
+      title = `${data[0].platform}: ${subtestName}`;
+      tableNames = [subtestName];
+      rowNames = [...new Set(data.map(item => item.test))].sort();
+    } else {
+      tableNames = [...new Set(data.map(item => item.name))].sort();
+      rowNames = [...new Set(data.map(item => item.platform))].sort();
+    }
+
+    const text = originalRevision
+      ? `${originalRevision} (${originalProject})`
+      : originalProject;
+
+    window.document.title =
+      title || `Comparison between ${text} and ${newRevision} (${newProject})`;
```

Comment:
Might be worth considering using [react-helmet](https://github.com/nfl/react-helmet) for this type of thing in the future.

Reply:
I'll look into it - thanks